src/Event.cc

```diff
@@ -97,27 +97,63 @@ Event::Event() {
 void Event::_init() {
     _keywords.insert(TYPE);
     _keywords.insert(EVENTTIME);
-    _keywords.insert(HOSTID);
-    _keywords.insert(RUNID);
+//    _keywords.insert(RUNID);
     _keywords.insert(STATUS);
     _keywords.insert(TOPIC);
     _keywords.insert(PUBTIME);
 }
 
 Event::Event(cms::TextMessage *msg) {
-    _init();
+    // _init();
+
+    vector<std::string>names = msg->getPropertyNames();
+    unsigned int i;
 
     _psp = processTextMessage(msg);
 
-    _psp->set(TYPE, msg->getStringProperty(TYPE));
-    _psp->set(HOSTID, msg->getStringProperty(HOSTID));
-    _psp->set(RUNID, msg->getStringProperty(RUNID));
-    _psp->set(STATUS, msg->getStringProperty(STATUS));
-    _psp->set(TOPIC, msg->getStringProperty(TOPIC));
-    _psp->set(EVENTTIME, msg->getLongProperty(EVENTTIME));
-    _psp->set(PUBTIME, msg->getLongProperty(PUBTIME));
-/*
-*/
+    for (i = 0; i < names.size(); i++)
+            _keywords.insert(names[i]);
```

Comment:
You MUST put braces around this line because it's too easy to make mistakes without them.  I think you're only allowed to go without braces if it's on the same line as the `for`.
Why are you iterating over numbers rather than iterating over the contents of `names` (i.e., using `for (auto n = names.begin(); n != names.end(); ++n)`)?